### ISSUE TITLE
[Bug]: [DatePicker] [A11y] combobox should be opened by down arrow key

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#30042]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 18.10 GB / 31.74 GB
  Browsers:
    Edge: Chromium (120.0.2210.61)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/solitary-tree-3r7vnj

### Bug Description

1. Navigate to the https://codesandbox.io/p/sandbox/solitary-tree-3r7vnj
2. navigate to "Select a date combo box"
3. Press down arrow key

## Actual Behavior
nothing happens 

## Expected Behavior
the calendar is opened 

This expectation comes from APG pattern:
https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/

as well others comboboxes
https://react.fluentui.dev/?path=/docs/components-combobox--default
opens with down arrow key

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
